### PR TITLE
Update Qthreads recipe to support latest package versions

### PR DESCRIPTION
**Pull request by janciesko**
Updates the `Qthreads` recipe to support versions `1.17` and `1.18`. Bumps requirement for `hwloc` to `2.x`.

### CODE REVIEW

**Comment by janciesko** [comment]
This should be ready.

File: var/spack/repos/builtin/packages/qthreads/package.py [hunk 1]
```diff
@@ -58,13 +62,15 @@ class Qthreads(AutotoolsPackage):
...
         "stack_size",
-        default=4096,
+        default=16384,
         description="Specify number of bytes to use in a stack",
         values=is_integer,
     )
 
     depends_on("hwloc@1.0:1", when="@:1.15 +hwloc")
-    depends_on("hwloc@1.5:2", when="@1.16: +hwloc")
+    depends_on("hwloc@1.5:2", when="@:1.16 +hwloc")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/qthreads/package.py:71]: Is this:
```suggestion
    depends_on("hwloc@1.5:2", when="@1.16 +hwloc")
```
what you meant?

File: var/spack/repos/builtin/packages/qthreads/package.py [hunk 2]
```diff
@@ -58,13 +62,15 @@ class Qthreads(AutotoolsPackage):
...
+    depends_on("hwloc@2", when="@:1.17 +hwloc")
+    depends_on("hwloc@2", when="@:1.18 +hwloc")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/qthreads/package.py:73]:
```suggestion
    depends_on("hwloc@2", when="@1.17: +hwloc")
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**